src/libs/auth.ts

```diff
@@ -97,71 +127,206 @@ const revokeTokens = async () => {
   }
 };
 
-const getSigninArgs = (includeBillingScope) => {
+const getSigninArgs = (includeBillingScope: boolean): ExtraSigninRequestArgs => {
   return Utils.cond(
-    [includeBillingScope === false, () => ({})],
+    [!includeBillingScope, (): ExtraSigninRequestArgs => ({})],
     // For B2C use a dedicated policy endpoint configured for the GCP cloud-billing scope.
-    () => ({
+    (): ExtraSigninRequestArgs => ({
       extraQueryParams: { access_type: 'offline', p: getConfig().b2cBillingPolicy },
       extraTokenParams: { p: getConfig().b2cBillingPolicy },
     })
   );
 };
 
 export const signIn = async (includeBillingScope = false): Promise<User> => {
-  const args = getSigninArgs(includeBillingScope);
-  const user: User = await getAuthInstance().signinPopup(args);
+  // we should handle if we get back null or false here (if loading the authTokenFails)
+  const authTokenState: AuthTokenState = await loadAuthToken(includeBillingScope, true);
```

Comment:
I'm fine with that change. Since this PR doesn't _introduce_ the term "auth token", if we do decide to change it, I would prefer that be done in a follow up PR.